server/aesdsocket.c

```diff
@@ -267,25 +289,47 @@ void* receive_send_data(void* thread_param)
 		close_all();
 		exit (EXIT_FAILURE);					
 	}	
-										
-	lseek(write_fd, 0, SEEK_SET); //set cursor to start
-	read_bytes = read(write_fd, thread_func_args->transmit_buffer, size); //read bytes
-	if(read_bytes < 0)
+	
+	
+    	
+	while((read_bytes = read(write_fd,&ch,1)) > 0)
 	{
-		close(thread_func_args->client_fd);
-		printf("Error in reading bytes\n");
-		close_all();
-		exit (EXIT_FAILURE);
-	}
+		if(read_bytes < 0)
+		{
+			close(thread_func_args->client_fd);
+			printf("Error in reading bytes\n");
+			close_all();
+			exit (EXIT_FAILURE);
+		}
 
-	send_bytes = send(thread_func_args->client_fd, thread_func_args->transmit_buffer, read_bytes, 0);//send bytes
-	if(send_bytes < 0)
-	{
-		close(thread_func_args->client_fd);
-		printf("Error in sending bytes\n");
-		close_all();
-		exit (EXIT_FAILURE);
-	}			
+		thread_func_args->transmit_buffer[index] = ch;
+
+		if(thread_func_args->transmit_buffer[index] == '\n')
+		{
+		    	write_size = (index + 1) - newline_index;
+		 	
+			send_bytes = send(thread_func_args->client_fd, thread_func_args->transmit_buffer+newline_index, write_size, 0);//send bytes
+			if(send_bytes < 0)
+			{
+				close(thread_func_args->client_fd);
+				printf("Error in sending bytes\n");
+				close_all();
+				exit (EXIT_FAILURE);
+			}
+		    
+		   	 newline_index = index + 1;
+
+		}
+
+		index++;
+
+		if(index >= newtransmit_buffer_size)
+		{
+		    newtransmit_buffer_size += BUFFER_SIZE;
```

Comment:
@midhun9 I think this logic is only going to work if you can malloc enough memory to hold all packets sent.